### PR TITLE
feat(plugin-blanks): arrow navigation

### LINKED ISSUE #3248: editor: blanks left and right arrow keys navigation
It's impossible to both enter and exit the blanks using left and right arrow keys.

### PR DESCRIPTION
**Pull request by hejtful**
For now, this PR includes:
- fix styling for answers drag-and-dropped into blanks
- refactoring (simplifying code, splitting up components)
- fixes https://github.com/serlo/frontend/issues/3248
- autofocusing blank input when adding a new blank
- focusing editor when removing a blank
- had to split the editor and static renderer components, to avoid bundle size increase (last commit). If you see a simpler way, please advise! 🙏

### CODE REVIEW

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx
Comment (elbotho) [inline packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx]: tiny thing: 
could we move the cursor to the end of the input when coming from the right and in front of the first character when coming from left?

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/index.tsx
Comment (elbotho) [inline packages/editor/src/plugins/fill-in-the-blanks-exercise/index.tsx]: just for testing or should this be merged?

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer-static.tsx
Comment (elbotho) [inline packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer-static.tsx]: (out of scope here, but arrow keys to move to the next blank might be a nice addition for users as well)

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx
Comment (elbotho) [inline packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx:31]: 🙏

**Review by elbotho — COMMENTED** [review]
simplifications look great.
ux improved a lot, thank you.
I added some comments. Most are take it or leave it.

> had to split the editor and static renderer components, to avoid bundle size increase (last commit). If you see a simpler way, please advise! 🙏

I don't mind the split in general. Maybe you could reuse the static input field in the `BlankRenderer` or create a new component just for the input field to avoid the almost complete duplication here (and render inconsistencies in the future).

**Comment by elbotho** [comment]
There will probably be some conflicts with my PRs from yesterday. Not sure, but since they are smaller it might still make sense to merge yours first.

**Comment by elbotho** [comment]
[preview with (messy) d&d and typing example](https://frontend-git-feat-fill-blanks-dragndrop-serlo.vercel.app/___editor_preview?state=%7B%22plugin%22%3A%22rows%22%2C%22state%22%3A%5B%7B%22plugin%22%3A%22exercise%22%2C%22state%22%3A%7B%22content%22%3A%7B%22plugin%22%3A%22rows%22%2C%22state%22%3A%5B%7B%22plugin%22%3A%22text%22%2C%22state%22%3A%5B%7B%22type%22%3A%22p%22%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%5D%2C%22id%22%3A%22b83512b9-7ba6-43b0-96b7-8fe8152e0d3a%22%7D%5D%2C%22id%22%3A%2242826ec1-bf68-42c7-bbc3-fa0d64e12c27%22%7D%2C%22interactive%22%3A%7B%22plugin%22%3A%22blanksExercise%22%2C%22state%22%3A%7B%22text%22%3A%7B%22plugin%22%3A%22text%22%2C%22state%22%3A%5B%7B%22type%22%3A%22p%22%2C%22children%22%3A%5B%7B%22text%22%3A%22asd+asd+%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%22e8031c1a-860b-404b-ae01-72b736c1b9c1%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22asd%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+as%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%229ac6dc61-f8a4-4ff6-8aa2-5c22bc62e027%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22d%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%221097178b-4b94-4bff-898c-e2db40defaf0%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22123%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+asdasd+%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%22f12e17e8-d59b-4ca4-801c-55ba87374ace%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22asd%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+asdas+d%22%7D%5D%7D%5D%2C%22id%22%3A%225d7c6265-b513-4211-9fdd-14124232e832%22%7D%2C%22mode%22%3A%22drag-and-drop%22%7D%2C%22id%22%3A%22cd8670db-173f-4d8d-8e80-f341c1188e42%22%7D%7D%2C%22id%22%3A%22b6293262-6451-497f-8b04-25113120815c%22%7D%2C%7B%22plugin%22%3A%22exercise%22%2C%22state%22%3A%7B%22content%22%3A%7B%22plugin%22%3A%22rows%22%2C%22state%22%3A%5B%7B%22plugin%22%3A%22text%22%2C%22state%22%3A%5B%7B%22type%22%3A%22p%22%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%5D%2C%22id%22%3A%22af1bb543-ac56-42bb-8fe4-ae063a72a5eb%22%7D%5D%2C%22id%22%3A%22af6ba856-e437-420e-ae6a-c9e6f03127b8%22%7D%2C%22interactive%22%3A%7B%22plugin%22%3A%22blanksExercise%22%2C%22state%22%3A%7B%22text%22%3A%7B%22plugin%22%3A%22text%22%2C%22state%22%3A%5B%7B%22type%22%3A%22p%22%2C%22children%22%3A%5B%7B%22text%22%3A%22asd+asd+asda+sdas+%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%225b236ef8-f846-4032-bf4b-bf122bc36706%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22dasd%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+asd+asdas+%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%22ab084775-9e1d-4d40-8f5a-4294c9f7b6e5%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22dasd%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+asd+asd+%22%7D%2C%7B%22type%22%3A%22textBlank%22%2C%22blankId%22%3A%22ebe080ac-a9d1-489a-9e06-2ae5e2a300ff%22%2C%22correctAnswers%22%3A%5B%7B%22answer%22%3A%22asdasd%22%7D%5D%2C%22children%22%3A%5B%7B%22text%22%3A%22%22%7D%5D%7D%2C%7B%22text%22%3A%22+%22%7D%5D%7D%5D%2C%22id%22%3A%2280d82f0a-3375-45f3-ade3-e6d41f5dda94%22%7D%2C%22mode%22%3A%22typing%22%7D%2C%22id%22%3A%22d09b2f51-a9fd-4a86-9fa2-1c947f06fe43%22%7D%7D%2C%22id%22%3A%22ae1d65df-452d-411f-b57e-61735f98f019%22%7D%5D%7D)

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/index.tsx
Comment (hejtful) [reply packages/editor/src/plugins/fill-in-the-blanks-exercise/index.tsx]: Great catch! 😅

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx
Comment (hejtful) [reply packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx]: Nice suggestion, I'll try to implement it.

For me, in Chrome, it actually does function like that, most of the time. But after your comment, I did notice inconsistencies, sometimes it will be at the end when coming from the left, so I guess that needs to be manually handled.

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer-static.tsx
Comment (hejtful) [reply packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer-static.tsx]: Could you maybe open an issue, explaining how you imagine that working? I guess we wouldn't just be using simple arrow left/right?

File: packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx
Comment (hejtful) [reply packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx]: Done :)

**Comment by hejtful** [comment]
@elbotho Thanks for the feedback! Updated the PR with your suggestions :)

**Review by elbotho — APPROVED** [review]
thank you. maybe wait for the review by @LarsTheGlidingSquirrel since he is way more familiar with the code :)

**Review by LarsTheGlidingSquirrel — APPROVED** [review]

**Pull request merged**
